packages/api/src/lib/job-portal-api.ts

```diff
@@ -0,0 +1,68 @@
+import axios from 'axios';
+import { getAuthHeaders } from './lms';
+import { OrganizationData, RecruiterData, StudentData } from './job-portal';
+
+export async function createStudentProfile(data: StudentData) {
+  await axios.post('/api/job-portal/create-student-profile', data, {
+    headers: getAuthHeaders(),
+  });
+}
+
+export async function createRecruiterProfile(data: RecruiterData) {
+  await axios.post('/api/job-portal/create-recruiter-profile', data, {
+    headers: getAuthHeaders(),
+  });
+}
+export async function upDateRecruiterProfile(data: RecruiterData) {
+  await axios.post('/api/job-portal/update-recruiter-profile', data, {
+    headers: getAuthHeaders(),
+  });
+}
+
+export async function getStudentProfile() {
+  const resp = await axios.get('/api/job-portal/get-student-profile', {
+    headers: getAuthHeaders(),
+  });
+  return resp.data as StudentData[];
+}
+
+export async function getRecruiterProfile() {
+  const resp = await axios.get('/api/job-portal/get-recruiter-profile', {
+    headers: getAuthHeaders(),
+  });
+  return resp.data as RecruiterData[];
+}
+
+export async function checkIfUserRegisteredOnJP(userId: string) {
+  const response = await axios.post('/api/job-portal/check-user-registered-on-jp', { userId });
+  return response.data as { exists: boolean };
+}
+
+export async function createOrganizationProfile(data: OrganizationData) {
+  await axios.post('/api/job-portal/create-organization-profile', data, {
+    headers: getAuthHeaders(),
+  });
+}
+export async function updateOrganizationProfile(data: OrganizationData, id: number) {
+  await axios.post(
+    '/api/job-portal/update-organization-profile',
+    { data, id },
+    { headers: getAuthHeaders() }
+  );
+}
+
+export async function getOrganizationProfile(id: number) {
+  const resp = await axios.get('/api/job-portal/get-organization-profile', {
+    headers: getAuthHeaders(),
+    params: { id },
+  });
+  return resp.data as OrganizationData[];
+}
+
+export async function getOrganizationId(organizationName: string) {
```

Comment:
Don't use this API. Using string name as identifiers is brittle. Manage the data flows so that the backend returns the `id` in `the OrganizationData`. When `id` is not available, use a different type. Eg. in for createOrganizationProfile, use a new type.

For an example see `CreateHomeworkRequest`:
https://github.com/slatex/sTeX-React/blob/02e6dc5a14b16d187153cbc6a40d86109136ba0c/packages/api/src/lib/homework-api.ts#L35